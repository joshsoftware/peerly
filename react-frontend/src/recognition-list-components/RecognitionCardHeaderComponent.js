import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Form } from "core-components/form/FormComponent";
import { Button } from "core-components/button/ButtonComponent";
import ImageComponent from "core-components/image/ImageComponent";

const ProfileImage = styled.div`
  width: 12vh;
`;

const MenuButton = styled(Button)`
  color: var(--black);
  background-color: var(--white);
  font-weight: bold;
  border: none;
`;

const RecognitionCardHeaderComponent = ({
  givenAt,
  givenForName,
  givenForImage,
  coreValue,
}) => {
  return (
    <div>
      <div className="d-flex justify-content-end">
        <MenuButton>...</MenuButton>
      </div>
      <div className="d-flex flex-row">
        <ProfileImage>
          <ImageComponent
            src={givenForImage}
            roundedCircle={true}
            alt="Profile"
          />
        </ProfileImage>
        <div className="text-start ml-2">
          <Form.Label className="font-weight-bold ">{givenForName}</Form.Label>
          <div>
            <Form.Label> got a high five for</Form.Label>
            <Form.Label className="ml-1 font-weight-bold">
              {coreValue}
            </Form.Label>
          </div>
          <Form.Label className="text-muted">{givenAt}</Form.Label>
        </div>
      </div>
    </div>
  );
};
RecognitionCardHeaderComponent.propTypes = {
  givenForName: PropTypes.string.isRequired,
  givenForImage: PropTypes.string,
  givenAt: PropTypes.string.isRequired,
  coreValue: PropTypes.string.isRequired,
};

RecognitionCardHeaderComponent.defaultProps = {
  givenForImage: "https://i.picsum.photos/id/2/200/200.jpg",
};

export default React.memo(RecognitionCardHeaderComponent);
