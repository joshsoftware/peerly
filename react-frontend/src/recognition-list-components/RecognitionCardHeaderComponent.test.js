import React from "react";
import { render } from "@testing-library/react";

import RecognitionCardHeaderComponent from "recognition-list-components/RecognitionCardHeaderComponent";

describe("Recognition card header Component test", () => {
  test("should equal snapshot", () => {
    const { asFragment } = render(
      <RecognitionCardHeaderComponent
        givenForName="Roy"
        coreValue="positive"
        givenAt="Today 10:30 AM"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
