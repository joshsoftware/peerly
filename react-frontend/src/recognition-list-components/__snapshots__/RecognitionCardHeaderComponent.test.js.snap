// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition card header Component test should equal snapshot 1`] = `
<DocumentFragment>
  .c1 {
  width: 12vh;
}

.c0 {
  color: var(--black);
  background-color: var(--white);
  font-weight: bold;
  border: none;
}

<div>
    <div
      class="d-flex justify-content-end"
    >
      <button
        class="c0 btn btn-primary"
        type="button"
      >
        ...
      </button>
    </div>
    <div
      class="d-flex flex-row"
    >
      <div
        class="c1"
      >
        <img
          alt="Profile"
          class="img-fluid rounded-circle"
          src="https://i.picsum.photos/id/2/200/200.jpg"
        />
      </div>
      <div
        class="text-start ml-2"
      >
        <label
          class="font-weight-bold  form-label"
        >
          Roy
        </label>
        <div>
          <label
            class="form-label"
          >
             got a high five for
          </label>
          <label
            class="ml-1 font-weight-bold form-label"
          >
            positive
          </label>
        </div>
        <label
          class="text-muted form-label"
        >
          Today 10:30 AM
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
